### ISSUE TITLE
Only works with PNG images, fails on JPG or WebP

### ISSUE THREAD

**Issue by wadeharrell** [burnpiro/puppeteer-screenshot-tester#47]
**Describe the bug**
Specified a webp first, then tried JPG (trying to keep image sizes small).  Both get the same error
```
    Error [ERR_UNHANDLED_ERROR]: Unhandled error. (TypeError: Cannot read property 'width' of null

      at node_modules/nodejs-resemble/resemble.js:121:27
      at Object.<anonymous>.exports.PNG.<anonymous> (node_modules/pngjs/lib/png.js:81:7)
```

**Please complete the following information:**
 - OS: macOS 11.4
 - Node 14.17.6

**To Reproduce**
```
  const tester = await ScreenshotTester(0.1);
  const result = await tester(testPage, 'test', {
    fullPage: true,
    type: 'webp',
    path: '/screenshots/test.webp',
  });
```
&
```
  const tester = await ScreenshotTester(0.1);
  const result = await tester(testPage, 'test', {
    fullPage: true,
    type: 'jpeg',
    path: '/screenshots/test.jpg',
  });
```
first run works and generates the image but second run throws an error

**Expected behavior**
Support all three screenshot formats supported by puppeteer https://github.com/puppeteer/puppeteer/blob/main/docs/api.md#pagescreenshotoptions

**Stack Trace**
```
    Error [ERR_UNHANDLED_ERROR]: Unhandled error. (TypeError: Cannot read property 'width' of null

      at node_modules/nodejs-resemble/resemble.js:121:27
      at Object.<anonymous>.exports.PNG.<anonymous> (node_modules/pngjs/lib/png.js:81:7)
      at Object.<anonymous>.module.exports.Object.<anonymous>.ParserAsync._handleError (node_modules/pngjs/lib/parser-async.js:36:8)
      at Object.<anonymous>.module.exports.Object.<anonymous>.Parser._parseSignature (node_modules/pngjs/lib/parser.js:53:12)
      at Object.<anonymous>.module.exports.Object.<anonymous>.ChunkStream._processRead (node_modules/pngjs/lib/chunkstream.js:174:13)
      at Object.<anonymous>.module.exports.Object.<anonymous>.ChunkStream._process (node_modules/pngjs/lib/chunkstream.js:193:14)
      at Object.<anonymous>.module.exports.Object.<anonymous>.ChunkStream.write (node_modules/pngjs/lib/chunkstream.js:61:8))
      at Object.<anonymous>.module.exports.Object.<anonymous>.ChunkStream._process (node_modules/pngjs/lib/chunkstream.js:207:10)
      at Object.<anonymous>.module.exports.Object.<anonymous>.ChunkStream.write (node_modules/pngjs/lib/chunkstream.js:61:8)
      at Object.<anonymous>.module.exports.Object.<anonymous>.ChunkStream.end (node_modules/pngjs/lib/chunkstream.js:74:10)
      at Object.<anonymous>.exports.PNG.Object.<anonymous>.PNG.end (node_modules/pngjs/lib/png.js:98:16)
      at Object.<anonymous>.exports.PNG.Object.<anonymous>.PNG.parse (node_modules/pngjs/lib/png.js:88:8)
      at loadImageData (node_modules/nodejs-resemble/resemble.js:120:9)
      at compare (node_modules/nodejs-resemble/resemble.js:504:4)
      at wrapper (node_modules/nodejs-resemble/resemble.js:582:7)
```

**Additional context**

**Comment by burnpiro** [comment]
Hi, the error is caused by `reasembleJS` and it's because it only supports `.png` files (I'm not sure if other formats with transparency are supported but neither `.jpg` nor `.webm` are one of them). You need transparency to create a diff, but I guess we could work with white or black backgrounds as well. I could try to rewrite that part of the code and replace transparency with a solid background. Let me look at that over the weekend.

**Comment by wadeharrell** [comment]
saw that, figured you knew the guy working on node-resemble.js ;)

Unfortunately I don't see any npm package for webp as straight forward as pngjs, but webp does support alpha https://developers.google.com/speed/webp/gallery2 so in theory it should be able to do everything you can with PNG at a much smaller file size; JPG is out of luck in that regard though.

In the context of puppeteer-screenshot-tester you know that you have chrome, so perhaps could leverage some canvas APIs? i.e. https://stackoverflow.com/a/55896125
I could see the case for even converting webp & jpg to PNG before sending them to resemble, then converting the diff to the desired output on the way back.

p.s. thanks for the quick response!

**Comment by burnpiro** [comment]
Fixed in [v1.5.0](https://github.com/burnpiro/puppeteer-screenshot-tester/releases/tag/v1.5.0).

Additionally, there is a new option to change the default compression levels for images. This might help with reducing the size if you want.
